### ISSUE TITLE
Question: Is there a way to properly set result for behave run?

### ISSUE THREAD

**Issue by modehnal** [behave/behave#1169]
I have a negated test, if it fails, that is good and is considered pass and vice versa.

I can easily solve that by making another step in python like this:
```python
try:
    # Execute the failing step.
    context.execute_steps(f'* Start application "{application}" via command "{command}"')
except Exception:
    # Expected exception.
    return
else:
    # Nothing went wrong.
    assert False, "This test was supposed to fail."
```

What I am wondering about is, if there is a way on behave side to properly do this without a need to do another step? I am unaware if behave itself supports this and I might have overlooked the feature.

I want to make this work with just a Tag - mark a test with a specific tag, and reverse logic for pass/fail. (I am already using the Tags for other useful stuff like '**autoretry=X**' that will run a test X times before failing or '**stability=Y**' to run a test Y times and all have to pass for final result to be pass)

I was somewhat successful. What I am doing is executing our common before/after scenario functions. In there I am checking if there is a Tag (in my case '\@\_\_negate\_\_') and based scenario.status I am changing the status.

What seems to work well is the following:
```python
scenario.steps[-1].status = Status.passed
scenario.clear_status()
scenario.set_status("passed")
scenario.feature.clear_status()
scenario.feature.set_status("passed")
```
That will make the final status report to provide my desired outcome. The issue that I am having though, is that I cannot find a way how to provide a return code for the entire run and in my attempts nothing has worked. So even if all parts are reported as Pass the return code is still 1.

Since we also work with the behave return code I cannot implement this feature without the return code being included. I thought about using an after_all hook and do sys.exit(0) but at that point I can break other stuff so I came with a question here. I realize I am hacking around a bit and if it is not possible than I will just figure out a logic via steps.

Thank you.

**Comment by bittner** [comment]
I may be wrong when judging your attempted solution, forgive me in that case, but shouldn't this be realized by a very simple test, like this?

```gherkin
When I start the application with a wrong command
Then the startup should fail
```

And in the "the startup should fail" BDD function you perform whatever `assert` necessary to evaluate the "failing" to `True`.

Messing around with test tooling is not only tedious but also difficult to understand for anyone taking over maintenance of your code after you. Hence, make all your tests as simple and straight-forward as possible. Really, really simple, extremely straight-forward.

Test suites grow and grow, they inevitably get out of control just like the business code (if without tests), otherwise.

**Comment by modehnal** [comment]
Yeah, the test case and solution are very simple just like you wrote.

I agree the simplicity is the key here.

I was just wondering if there is something like this in behave but I guess not and as you said it makes more sense to write it as a separate steps.

I also agree that messing with tooling can be tedious and in hindsight the solution should not be making a tool to mark a test case, but making the test case straight-forward. I will be dropping the attempt to make feature like this.

Thank you
